src/main/java/com/atlassian/bitbucket/jenkins/internal/provider/DefaultJenkinsProvider.java

```diff
@@ -4,7 +4,17 @@
 
 public class DefaultJenkinsProvider implements JenkinsProvider {
 
+    private volatile String jenkinsRootUrl;
```

Comment:
why do we need to cache this lookup?

Reply:
Implementation of getUrl is non-trivial. The existing method iterates over existing extensions and fetches the Jenkins URL. Since the Jenkins URL would be needed across plugin, this was stored.

Reply:
I remain unconvinced. By default there is a configured url, so unless that has been actively removed it is a rather simple look up. I do not like caching this because it is so rarely needed and seems like just overhead.